File: packages/localization/android/src/main/java/io/capawesome/capacitorjs/plugins/localization/classes/results/GetSettingsResult.java
package io.capawesome.capacitorjs.plugins.localization.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.localization.interfaces.Result;

public class GetSettingsResult implements Result {

    private final int firstDayOfWeek;

    @NonNull
    private final String timeZone;

    private final boolean uses24HourClock;

    public GetSettingsResult(@NonNull String timeZone, boolean uses24HourClock, int firstDayOfWeek) {
        this.timeZone = timeZone;
        this.uses24HourClock = uses24HourClock;
        this.firstDayOfWeek = firstDayOfWeek;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("firstDayOfWeek", firstDayOfWeek);
        result.put("timeZone", timeZone);
        result.put("uses24HourClock", uses24HourClock);
        return result;
    }
}

File: packages/localization/ios/Plugin/Classes/Results/GetSettingsResult.swift
import Foundation
import Capacitor

@objc public class GetSettingsResult: NSObject, Result {
    let firstDayOfWeek: Int
    let timeZone: String
    let uses24HourClock: Bool

    init(timeZone: String, uses24HourClock: Bool, firstDayOfWeek: Int) {
        self.timeZone = timeZone
        self.uses24HourClock = uses24HourClock
        self.firstDayOfWeek = firstDayOfWeek
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["firstDayOfWeek"] = firstDayOfWeek
        result["timeZone"] = timeZone
        result["uses24HourClock"] = uses24HourClock
        return result as AnyObject
    }
}
